apollo-ios-pagination/Sources/ApolloPagination/GraphQLQueryPager.swift

```diff
@@ -310,33 +495,110 @@ extension GraphQLQueryPager {
           shouldUpdate = true
         }
         let variables = query.__variables?.values.compactMap { $0._jsonEncodableValue?._jsonValue } ?? []
+        nextPageVarMap[variables] = nextPageData
+
+        if let latest {
+          let (previousPages, firstPage, nextPage) = latest
+          let value: Result<Output, Error> = .success(
+            Output(
+              previousPages: previousPages,
+              initialPage: firstPage,
+              nextPages: nextPage,
+              updateSource: data.source == .cache ? .cache : .fetch
+            )
+          )
+          if isLoadingAll {
+            queuedValue = value
+          } else {
+            currentValue = value
+          }
+        }
         if shouldUpdate {
           publisher.send(completion: .finished)
         }
-        varMap[variables] = nextPageData
+      case .failure(let error):
+        currentValue = .failure(error)
+        publisher.send(completion: .finished)
+      }
+    }
+
+    private func onPreviousFetch(
```

Comment:
If we add a `PaginationDirection` enum (as suggested above), we could probably combine `onSubsequentFetch` and `onPreviousFetch` into a single function that takes in a `direction`.